freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -98,72 +120,232 @@ def _get_requests_session(cls, retry_options={}):
         session.mount("https://", adapter)
         return session
 
-    def _get_auto_rebuild_image_list(self, event):
+    def _get_auto_rebuild_image_mapping(self):
         """
-        Get rebuild image list which can be auto rebuilt.
+        Get rebuild image mapping which images can be auto rebuilt.
 
-        :param event FlatpakModuleAdvisoryReadyEvent: The event this handler
-            is currently handling.
-        :rtype: list
-        :return: List of images which can be auto rebuilt.
+        :rtype: dict
+        :return: A dict which key is the image which can be rebuilt
+            and value are modules which are enabled in this image.
         """
         if not conf.pyxis_server_url:
             raise ValueError("'PYXIS_SERVER_URL' parameter should be set")
         self._pyxis = Pyxis(conf.pyxis_server_url)
 
         if not conf.flatpak_server_url:
-            raise ValueError(
-                "'FLATPAK_SERVER_URL' parameter should be set"
-            )
-
-        errata = Errata()
-        module_nvrs = errata.get_cve_affected_build_nvrs(event.advisory.errata_id, True)
-        rebuild_images_list = list()
+            raise ValueError("'FLATPAK_SERVER_URL' parameter should be set")
 
+        image_modules_mapping = {}
         with koji_service(
             conf.koji_profile, log, login=False, dry_run=self.dry_run
         ) as session:
-            for module_nvr in module_nvrs:
-                mmd = session.get_modulemd(module_nvr)
+            for advisory_module_nvr in self.advisory_module_nvrs:
+                mmd = session.get_modulemd(advisory_module_nvr)
                 content_index_url = "{}/released/contents/modules/{}:{}.json".format(
                     conf.flatpak_server_url,
                     mmd.get_module_name(),
                     mmd.get_stream_name(),
                 )
                 req_session = self._get_requests_session()
```

Comment:
This should initialized once outside of the loop.

Reply:
advisory_module_nvr is different, so every advisory_module_nvr will be initialized, right?

Reply:
What I meant is that creating `requests.Session()` every loop iteration should be avoided.

Reply:
Oh. You mean line "req_session = self._get_requests_session()"? I was thinking the difference parts.

Reply:
Yeah, it something we missed in previous PR.